### ISSUE TITLE
create-nx-workspace not compatible with windows filesystem

### ISSUE THREAD

**Issue by sebastiangug** [nrwl/nx#18677]
### Current Behavior

When attempting to run npx create-nx-workspace@latest test on a windows machine, in our case version `16.7.1`

It will eventually error out with this: 

Fetching @nx/nest...

 >  NX   generateFiles: No files found in "/Users/myusername/AppData/Local/Temp/tmp-13492-63PtD20hzABS/node_modules/@nx/js/src/generators/init/files". Are you sure you specified the correct path?
 
 In this case part of the `@nx/nest` executor, but I've seen reports of different plugins suffering from the same issue. Everything with the same exact versions works perfectly fine on macOS (tested it myself, I have both). 
 
 if I navigate to that file path all there is a `__filename__` file of 1kb, likely empty.
 
 My assumption is either the path is wrong for not including the drive letter `C:\` or by not adding the right files in there in the first place, don't know enough about the inner works of the `create-nx-workspace` command to comment further. 

### Expected Behavior

create-nx-workspace should be able to find the files required to fulfil its basic purpose on windows machines.

### Steps to Reproduce

1.install nx@16.7.1 globally
2. run `npx create-nx-workspace test`
3. choose node
4. choose nest
5. choose integrated monorepo
6. choose no dockerfile
7. choose yes cloud build
8. wait for the runner to install stuff
9. error

### Nx Report

```shell
Node   : 18.17.0
   OS     : win32-x64
   npm    : 9.8.1

   nx (global)    : 16.7.1
   nx             : 16.7.1
   @nx/js         : 16.7.1
   @nx/jest       : 16.7.1
   @nx/linter     : 16.7.1
   @nx/workspace  : 16.7.1
   @nx/devkit     : 16.7.1
   @nx/node       : 16.7.1
   @nrwl/tao      : 16.7.1
   nx-cloud       : 16.4.0-beta.1
```

### Failure Logs

```shell
added 508 packages, and audited 509 packages in 6s

44 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

>  NX  Generating @nx/workspace:preset

Fetching @nx/nest...

 >  NX   generateFiles: No files found in "/Users/Seb/AppData/Local/Temp/tmp-13492-63PtD20hzABS/node_modules/@nx/js/src/generators/init/files". Are you sure you specified the correct path?

   Pass --verbose to see the stacktrace.


 >  NX   Workspace creation failed, see above.

   Pass --verbose to see the stacktrace.
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by Cammisuli** [comment]
I just gave it a run on my machine, and things installed correctly. So lets find out why.
1. Whats your Windows version?
2. What terminal/shell are you using?

Here's mine: 
```
Edition	Windows 11 Home
Version	22H2
Installed on	‎2023-‎01-‎28
OS build	22621.2134
Experience	Windows Feature Experience Pack 1000.22659.1000.0
```
And I'm using powershell core:
```
pwsh --version
PowerShell 7.3.6
```

**Comment by Azuky** [comment]
@Cammisuli, I'm experiencing the same issue. It appears we're using the same Windows and PowerShell versions.

Windows:
```
Edition	Windows 11 Pro
Version	22H2
Installed on	‎2023-‎02-‎20
OS build	22621.2134
Experience	Windows Feature Experience Pack 1000.22659.1000.0
```

Powershell:
```
pwsh --version
PowerShell 7.3.6
```

The only distinction I can identify is that I'm attempting to create the workspace on a different disk (disk E: in my case). Could this be the cause?

**Comment by Cammisuli** [comment]
Hmm that could be a cause. Let me try with a different drive letter for you @Azuky

**Comment by Cammisuli** [comment]
Yup! Trying to do this on a separate drive letter causes the issue. I'll look into it.
